### DISCUSSION TITLE
Brief lag in beatmaps which start at beginning of audio file

### DISCUSSION THREAD

**Discussion by jthistle** [ppy/osu#14905] [Q&A]
[System specs](https://gist.github.com/jthistle/da7e7b0664fec11b90190cf11cf66a0b).
[osu! logs](https://gist.github.com/jthistle/72b013fddb0d0561a461d02e87a0f5f6).

When playing beatmaps that begin at or near the start of the audio file, there is a very brief, but noticeable, moment of lag. It's hard to see, but you can certainly hear it (on the second note of the first beatmap). It's at its most noticeable when playing the map, since it makes it very hard to get the timing right. 

Both plays in the video are automated so they should be perfectly timed. There is no lag in beatmaps that start more than a second or so after the beginning of the audio file.

Turn sound on:

[Video](https://user-images.githubusercontent.com/8274049/135470639-f2825d32-ec3d-4b3a-ae69-ec89439ab709.mp4)

In the video I've used two custom beatmaps to demonstrate the problem, but this happens consistently with every other beatmap I've tried. [The two demo beatmaps are available here](https://github.com/ppy/osu/files/7260137/lag_bug_beatmaps.zip).
 
I feel like this is an important bug, since it makes some maps basically un-fcable.

**Comment by peppy** [comment]
@jthistle can you show what this looks like with the ctrl-f11 graph view visible (hit ctrl-f11 twice)?

From the video it looks like this is not a full game lag but just the audio/clock stopping for a moment, is that what you are feeling?

**Reply by jthistle** [reply]
Will rerecord the video with this on later once I'm home.

As for what it feels like, it feels as is if the game clock slows down for a brief moment, and then actually runs too fast again for a similar amount of time to catch up.

There is one interesting development here: since recording this, I have changed my PulseAudio latency and priority settings, in an attempt to get lower latency on stable  Whatever I did works, and I now get lazer-compatible latency running stable through Wine. But, and this might be placebo, sometimes it feels like I now get a similar affect in stable to the one described in this issue. I'll have to check this properly.

What I'm saying is, this could easily be an issue with PulseAudio / some other Linux subsystem. More investigation is needed.

Also note that the changes I have made will affect the video that I make later in comparison to before. I'd be surprised if the issue were gone, though.

**Reply by peppy** [reply]
Oh right. It's a linux issue. That changes everything.

**Reply by jthistle** [reply]
It _might_ be a Linux issue. When you say it changes everything - is Lazer actually supporting Linux, or is it more just a 'eh, we give you a build for it but you can fix any problems yourself'?

**Reply by peppy** [reply]
Linux is a wildcard, we expect users to fix issues themselves in most cases. Not trying to be dismissive, but we just don't have the manpower to make things work with every window manager, every audio backend, every kernel etc.

If the issue reproduces on all platforms then it's something we would look at as a priority, but I can't seem to repro what you're seeing at my end. My guess is that there's some latency with starting a track playback, and you're seeing this because the track is starting right next to where gameplay is starting.

**Reply by jthistle** [reply]
Cool, I'll leave this alone for now then. May even try building lazer myself and seeing if I can fix it - the joys of open source game dev!
